test-project/helloWorld/build.gradle

```diff
@@ -3,10 +3,14 @@ repositories {
     mavenCentral()
 }
 
+def target = System.getProperty("substrate.target")
+
 dependencies {
-    implementation 'com.googlecode.plist:dd-plist:1.22'
-    implementation 'com.github.jnr:jnr-ffi:2.1.10'
-    implementation 'org.bouncycastle:bcpkix-jdk15on:1.49'
+    if ("ios" == target) {
+        implementation 'com.googlecode.plist:dd-plist:1.22'
+        implementation 'com.github.jnr:jnr-ffi:2.1.10'
+        implementation 'org.bouncycastle:bcpkix-jdk15on:1.49'
```

Comment:
why is the test project dependent on this?

Reply:
The test-project doesn't have a dependency on substrate, because tests are run before building the jar. We just add manually the path to substrate/build/classes (and resources), as these are compiled before running tests.

Reply:
do you mean those 3 lines can be removed?

Reply:
I didn't say that, quite the contrary: Since we don't have a dependency on substrate, its transitive dependencies aren't resolved, and that's why we have to add them.